### DISCUSSION TITLE
Create new fieldtype for config

### DISCUSSION THREAD

**Discussion by ArMaNKashef** [joomengine/Joomla-Component-Builder#623] [Q&A] [answered]
Hi.
I created [Templatestyle](https://docs.joomla.org/Special:MyLanguage/Templatestyle_form_field_type) field for adding in config of my component. but when add add a field with this type nothing show in config.
this is my field Properties:
![image](https://user-images.githubusercontent.com/27265318/102706282-b5ad4280-42a5-11eb-8d89-5fef43191720.png)
please help me
thanks

**Comment by ManfredHuebner** [comment]
Hi, as far as I know it is not possible to "create" a fieldtype in that way. In field "type" only official joomla fieldtypes are allowed. Field types by individual names must be from type custom and represent lists or radio buttons. What you can do is allowing Joomla custom fields as part of your admin view and then create in your component a Joomla custom field for that admin view.
Regards
Manfred

**Reply by ArMaNKashef** [reply]
Hi Manfred
Thanks for your answering
Templatestyle is an official Joomla field and you can see it on this link:
https://docs.joomla.org/Special:MyLanguage/Templatestyle_form_field_type
When manually added this XML in config.xml it worked correctly 
`  <field name="course_templatestyle"
    type="templatestyle"
    client="site"
    description="Some description"
    label="Template Style" >
      <option value="">JOPTION_USE_DEFAULT</option>
  </field>`
but I don't want to hardcode for doing this ,I want to add this by JCB.

**Reply by ManfredHuebner** [reply]
You can try to create a fieldtype in JCB but I think it will not work as I cannot find a class definition of a type "templatestyle" in Joomla.

**Reply by ArMaNKashef** [reply]
Class definition of themplatestyle -> https://github.com/joomla/joomla-cms/blob/a80d9d9e63d44faea6ee1d57185769beea420b26/libraries/src/Form/Field/TemplatestyleField.php

**Comment by ManfredHuebner** [comment]
Please watch the videos on Joomlamount youtube channel regarding field types for more information:
https://www.youtube.com/c/JoomlaMount/search?query=fieldtype

**Comment by Llewellynvdm** [comment, marked as answer]
Does JCB add the xml for the field? it should, and it should work.

JCB was made to extend and adding new field types that have not already been added should work.

**Reply by ManfredHuebner** [reply]
HI Llewellyn, I tried to create a field type "templatestyle" and a field of this field type.

But neither XML or in the database anything will be created. Any idea what I am doing wrong?

**Reply by ManfredHuebner** [reply]
I got it now.

It is because of the static definition of "Templatestyle" as a special field in line 3668 (fieldGroups = array) of componentbuilder.php

For this type there is no implementation in c_Fields.php and that's why you cannot create the field ;-)

For me it is not an issue but ArMaNKashef will not be able to use it.

**Reply by Llewellynvdm** [reply]
try to move it from the [special array](https://github.com/vdm-io/Joomla-Component-Builder/blob/staging/admin/helpers/componentbuilder.php#L3698) to the [plain array](https://github.com/vdm-io/Joomla-Component-Builder/blob/staging/admin/helpers/componentbuilder.php#L3677), and let me know if it works.

**Reply by ManfredHuebner** [reply]
Yes that works :-)

**Reply by Llewellynvdm** [reply]
I have pushed the changes to the [staging branch](https://github.com/vdm-io/Joomla-Component-Builder/issues/630), and consider this resolved.
